hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/operators/common/HazelcastOperandTypeCheckerAware.java

```diff
@@ -70,10 +70,12 @@ default HazelcastCallBinding prepareBinding(SqlCallBinding binding) {
             RelDataType[] operandTypes = new RelDataType[binding.getOperandCount()];
             Arrays.fill(operandTypes, unknownType);
 
-            operandTypeInference.inferOperandTypes(binding, binding.getValidator().getUnknownType(), operandTypes);
+            if (operandTypeInference != null) {
```

Comment:
In which cases an `operandTypeInference` could be `null`? Generally, this should never happen, because otherwise, we may miss some inferred types of the children nodes, as explained in the JavaDoc.

Reply:
Fixed - it's again non-null.